R_scripts/Recursive_annot_enhanced.R

```diff
@@ -0,0 +1,167 @@
+# Helper function for QC to filter splice junctions based on read size and number of unique reads.
+filter_splicejunct_QC <- function(sj_df, mx_sj_overhang = 20, nr_of_uniqreads = 5) {
+  sj_df %>%
+    filter(V9 >= mx_sj_overhang, V7 >= nr_of_uniqreads)
+}
+
+annotate_splice_junctions <- function(sj_tab_folder        = cwd,
+                                      defined_introns_folder = str_glue("{cwd}/defined_introns.bed"),
+                                      sj_annot_out_folder  = str_glue("{cwd}/sj_annot_out_folder")) {
+   
+
+    # Check if the output_directory exists, and create it if it doesn't
+  if (!dir.exists(sj_annot_out_folder)) {
+    dir.create(sj_annot_out_folder, recursive = TRUE)
+    cat("Created output directory.\n")
+  } else {
+    cat("Output directory already exists.\n")
+  }
+  
+  defined_introns_file <- str_glue("{defined_introns_folder}/*intron.bed")
+  
+  message("Extracting splice junction files...")
+  
+  # Function to process and add extracted filename part to each dataframe
+  process_splice_file <- function(file) {
+    # Read the file
+    data <- fread(file, header = FALSE)
+
+    data <- filter_splicejunct_QC(data)
+
+    # Extract the part of the filename before .out.tab and add as a new column
+    filename_part <- sub("\\.out\\.tab$", "", basename(file))
+    data <- mutate(data, filename = filename_part)
+    return(data)
+  }
+  
+  # Extract all splice junction files
+  splice_files <- list.files(
+    path = sj_tab_folder,
+    pattern = "*.out.tab",
+    recursive = TRUE,
+    full.names = TRUE
+  )
+  
+  message("Processing splice junction files...")
+  
+  # Process all splice junction files
+  splice_data <- lapply(splice_files, process_splice_file) %>%
+    bind_rows()
+  
+  splice_data <- splice_data %>%
+    mutate(supported_by_samples = 1) %>%
+    group_by(V1, V2, V3, V4, V5, V6) %>%
+    summarise(
+      supported_by_samples = sum(supported_by_samples),
+      max_unique_mapping_count = max(V7),
+      min_multi_mapping_count = min(V8),
+      max_overhang = max(V9),
+      filename = paste(unique(filename), collapse = "; ")
+    ) %>%
+    mutate(supported_by_samples = supported_by_samples / length(splice_files))
+  
+
+  # Write the filtered splice junctions to an output file
+  filterd_sj_outfile_name <- str_glue("{sj_annot_out_folder}/all_samples_sj.tsv")
+  write_tsv(splice_data, filterd_sj_outfile_name, col_names = FALSE)
+  
+  message("Running bedtools intersect...")
+  # Intersect the splice junction file with the intron file using system command
+  intersect_command <- str_glue("bedtools intersect -wa -wb -a {filterd_sj_outfile_name} -b {defined_introns_file} > {sj_annot_out_folder}/intersect_sj_measured_vs_reference.tsv")
```

Comment:
1) Check out same wrappers that allows usage of bedtools directly in R (e.g,: https://cran.r-project.org/web/packages/bedr/vignettes/Using-bedr.html or https://github.com/PhanstielLab/bedtoolsr)
2) Do the intersect as stranded (-s parameter). Also be careful if both of the input files adhere to the .bed format specification (see e.g. https://www.ensembl.org/info/website/upload/bed.html)